### DISCUSSION TITLE
How to add new instances to the nats cluster?

### DISCUSSION THREAD

**Discussion by Doslin** [nats-io/nats-server#3759] [Q&A]
[I see that TiKV can be scaled up or down using the cli command](https://docs.pingcap.com/tidb/v6.1/scale-tidb-using-tiup#scale-out-a-tidbpdtikv-cluster), is there a similar operation on natscli?
or do you have a recommended operation ?

**Comment by Jarema** [comment]
You scale NATS Cluster by running another nats-server instance with `routes` directed at other node/nodes.

Assuming you have nats-server listening on cluster 4248:
`nats-server -p 4222 -cluster nats://localhost:4248 --cluster_name test-cluster`

You can add an additional node by running another server with `routes` targeting one of the other node's `cluster` addresses:
`nats-server -p 5222 -cluster nats://localhost:5248 -routes nats://localhost:4248 --cluster_name test-cluster`
`nats-server -p 6222 -cluster nats://localhost:6248 -routes nats://localhost:4248 --cluster_name test-cluster`

You can achieve the same by using a config file.

docs: https://docs.nats.io/running-a-nats-service/configuration/clustering
cluster config docs: https://docs.nats.io/running-a-nats-service/configuration/clustering/cluster_config

**Reply by Doslin** [reply]
What if I configure authorization?
How do I get other machines in the cluster to pass on the authorization?

**Reply by Jarema** [reply]
Refer the docs and authorization block:
https://docs.nats.io/running-a-nats-service/configuration/clustering/cluster_config
